front/src/applications/editor/tools/rangeEdition/speedSection/SpeedSectionEditionLayers.tsx

```diff
@@ -72,17 +75,29 @@ export const SpeedSectionEditionLayers = () => {
     return res;
   }, [interactionState, hoveredItem, entity, selectedSwitches]);
 
-  const highlightedTracksFeature: FeatureCollection = useMemo(() => {
+  const makeTrackRangeFeatures =
+    (flatEntity: EditorEntity) => (range: ApplicableTrackRange, i: number) => {
```

Comment:
I'm not fond of functions that return functions, because I think we loose in clarity. Can you transform this function like this ?

Also you can maybe rename it extractTrackRangeFeatures, what do you think ?
```suggestion
  const extractTrackRangeFeatures =
    (flatEntity: EditorEntity, range: ApplicableTrackRange, i: number) => {
```

Reply:
partially applied functions are a really common pattern that can be useful. 
Here flatMap expects a `(range: ApplicableTrackRange, i: number) => {}` and that’s what the function `makeTrackRangeFeatures(flatEntity)` is.
you can see this as a parametrized calledback. `flatEntity` is not a parameter.

`flatMap((range: ApplicableTrackRange, i: number) => makeTrackRangeFeatures(flatEntity, range, i))` is adding yet another function just to pass parameters.

I can modify it if you really want to (we can discuss more in our weekly front meeting) but I think It’s a good practice to get used to functional style patterns (which all of the `array.prototype.` tools are coming from, `.map`, `.flatMap`, `.filter`, `.reduce` are all higher order functions)

Reply:
I renamed the function

Reply:
I understand that partially applied functions are indeed a common and useful pattern, but I believe we should use them wisely since it can add unnecessary complexity.

Here, I think the partially applied function might be overcomplicating the implementation, especially because the function is not very complex and because the function is not very general. It makes the code more difficult to read and I think that the infra editor is complex enough, we should try to simplify the code as much as we can.

We can discuss it tomorrow if you want, since indeed partially applied functions can be really useful sometimes, but I think that here, we should remove it

Reply:
done